dlpy/network.py

```diff
@@ -726,22 +745,68 @@ def load_weights_from_file(self, path, format_type='KERAS'):
             contains the weight table.
         format_type : KERAS, CAFFE
             Specifies the source framework for the weights file
+        data_spec: list of :class:`DataSpec`, optional
+            data specification for input and output layer(s)            
 
         '''
         cas_lib_name, file_name = caslibify(self.conn, path, task='load')
 
-        self._retrieve_('deeplearn.dlimportmodelweights', model=self.model_table,
-                        modelWeights=dict(replace=True,
-                                          name=self.model_name + '_weights'),
-                        formatType=format_type, weightFilePath=file_name,
-                        caslib=cas_lib_name)
+        if (data_spec):
+            from io import StringIO
+            import sys
+
+            # redirect stderr
+            old_stderr = sys.stderr
```

Comment:
I'm not sure why this remapping of stderr is here.  That's kind of a sketchy thing to do and I'm not sure what it is accomplishing here.  The message level of the action is set to none, so it shouldn't be sending anything back as it is.

Reply:
I didn't like doing this but it was the only way to suppress the "ERROR: ..." returned by the action when the dataSpec parameter was not defined.  Despite setting the messagelevel to NONE, the error was always returned.  I tried a number of things to suppress the error, including setting the CAS session option for suppressing messages and nothing worked.

Reply:
It may be that there's a bug in the action so that it's still sending the error messages back.  However, I would think that using a context manager like the following would prevent anything from being printed:
```
with swat.option_context(print_messages=False):
    ... code with messages to suppress ...
```

Reply:
Ethem just walked over and made the same suggestion.  I tested it and it seems to work so I'm replacing the redirection of stderr with the context manager.